requests_kerberos/kerberos_.py

```diff
@@ -285,6 +309,22 @@ def deregister(self, response):
         response.request.deregister_hook('response', self.handle_response)
 
     def __call__(self, request):
+        if self.force_preemptive:
+            # add Authorization header before we receive a 401
+            # by the 401 handler
+            host = urlparse(request.url).hostname
+
+            with SniffLog(log) as sl:
+                auth_header = self.generate_request_header(None, host, is_preemptive=True)
+
+                if auth_header is None:
+                    # GSS setup failure, return error and details, if any
+                    raise KerberosExchangeError("Kerberos GSS Exchange failed: %s" % str(sl.stored_logs))
+
+            log.debug("HTTPKerberosAuth: Preemptive Authorization header: {0}".format(auth_header))
```

Comment:
Eh, if we're doing it already let's just leave it. I have really mixed feelings about it, but...we're probably ok.